packages/astro/src/vite-plugin-build-html/index.ts

```diff
@@ -138,6 +140,10 @@ export function rollupPluginAstroBuildHTML(options: PluginOptions): VitePlugin {
               const src = getAttribute(node, 'src');
               if (src?.startsWith(srcRoot) && !astroAssetMap.has(src)) {
                 astroAssetMap.set(src, fs.readFile(new URL(`file://${src}`)));
+              } else if (src?.startsWith(srcRootWeb) && !astroAssetMap.has(src)) {
+                const resolved = new URL('.' + src, astroConfig.projectRoot);
+                const assetId = viteifyURL(resolved);
+                astroAssetMap.set(src, fs.readFile(new URL(`file://${assetId}`)));
```

Comment:
```suggestion
                astroAssetMap.set(src, fs.readFile(resolved));
```
with the change to viteifyURL this may be better